grpc/src/test/java/com/linecorp/armeria/server/grpc/UnframedGrpcServiceTest.java

```diff
@@ -112,8 +112,11 @@ public void emptyCall(Empty request, StreamObserver<Empty> responseObserver) {
         });
         final HttpResponse response = unframedGrpcService.serve(ctx, request);
         final AggregatedHttpResponse res = response.aggregate().get();
-        assertThat(res.status()).isEqualTo(HttpStatus.OK);
-        assertThat(res.content().isEmpty()).isTrue();
+        assertThat(res.status()).isEqualTo(HttpStatus.CLIENT_CLOSED_REQUEST);
+        assertThat(res.contentUtf8())
+                .isEqualTo("http-status: 499, Client Closed Request\n" +
+                           "Caused by: \n" +
+                           "grpc-status: 1, CANCELLED, Completed without a response");
```

Comment:
Could you explain why 499 is an expected response? It looks like the client sent a valid request. Shouldn't the status code be 5xx?

Reply:
Initially, I tried to send a 500 Internal Server Error by fixing `GrpcStatus.grpcCodeToHttpStatus`.
https://github.com/line/armeria/blob/5b384fbe27e7e6f9225d6db91cbb684d09dfbb5e/grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcStatus.java#L177-L181
There were two concerns when I implemented this.
1. A custom mapping table would make a gap with upstream's mapping. Envoy might transcode this gRPC response to 499 HTTP status.
2. If we implement the status mapping with the existing API, it would be hard-coded and difficult to manage.
```
case CANCELLED:
    if ("Completed without a response".equals(status.getDescription()) {
        return HttpStatus.INTERNAL_SERVER_ERROR;
    }
    // 499 Client Closed Request is already used even if a request is timed out by a server-side, 
    // We also need to fix this together
    if ("Request timed out".equals(status.getDescription()) {
        return HttpStatus.INTERNAL_SERVER_ERROR;
    }
    ...
    return HttpStatus.CLIENT_CLOSED_REQUEST;
```
If we decide to make our own mapping table, we might need a general way of customizing the status mapping table.
For example, an expected HTTP status could be set to a request context's attributes when closing a call.

After thing about this and that, I wasn't so sure that it is the right approach to add our own mapping table.

Reply:
On second thought, perhaps, we could use `HttpStatusException` for the custom status mapping.
```java
Status status =
        Status.CANCELLED
              .withDescription("Completed without a response").
              .withCause(HttpStatusException(HttpStatus.INTERNAL_SERVER_ERROR);

public static HttpStatus grpcStatusToHttpStatus(Status grpcStatus) { 
   if (grpcStatus.getCause() instanceof HttpStatusException) {
        return ((HttpStatusException) grpcStatus.getCause()).httpStatus();
   } else {
       return grpcStatusCodeToHttpStatus(grpcStatus.getCode());
   }
}
```